194.1 trycatchcleanup alert modal ex createexampage

ArturZsigmond opened the pull request.

ArturZsigmond committed:
fix: moved the base implementation of the RaiseErrorMessage and the EventHandler in the ViewModelBase.cs, overrideing it in AdminBaseViewModel, since some viewmodels inherit one of them, but not just one. thus, since Admin inherits viewmodelbase, viewmodelbase is the best place to put it

ArturZsigmond committed:
feat: added RaiseErrorMessage and eventhandler to baseviewmodel.cs, since this is also used in some places and I don't want to dip into changing inheritance now

ArturZsigmond committed:
fix: added RaiseErrorMessage

ArturZsigmond committed:
fix: added try-catch for CourseViewModel and CoursePage with RaiseErrorMessage method

ArturZsigmond committed:
fix: added try-catch for CreateExamView and Page with RaiseErrorMessage method

ArturZsigmond committed:
fix: added try-catch block to CreateQuizView and page with RaiseErrorMessage method

ArturZsigmond committed:
fix: added try-catch block

ArturZsigmond committed:
fix: added try-catch blocks

ArturZsigmond committed:
fix: added try-catch blocks

ArturZsigmond committed:
fix: added try-catch blocks

ArturZsigmond committed:
fix: added try-catch blocks

ArturZsigmond committed:
fix: added try-catch blocks

ArturZsigmond committed:
fix: added try-catch blocks

UjfalusiAbel commented on Duo/Views/CoursePage.xaml.cs:
```diff
@@ -51,16 +80,14 @@ private async void ModulesListView_ItemClick(object sender, ItemClickEventArgs e
-                    CloseButtonText = "OK"
-                };
+
+                // Use RaiseErrorMessage instead of inline dialog
```
Delete unnecessary 1 line comments

UjfalusiAbel commented on Duo/Views/CoursePage.xaml.cs:
```diff
@@ -22,17 +24,44 @@ public CoursePage()
                 ModulesListView.ItemClick += ModulesListView_ItemClick;
                 vm.StartCourseProgressTimer();
+
+                // Subscribe to error message event
```
Delete unnecessary 1 line comments

UjfalusiAbel commented on Duo/Views/Pages/EndQuizPage.xaml.cs:
```diff
@@ -80,15 +49,54 @@ private void CloseButton_Click(object sender, RoutedEventArgs e)
+            await dialog.ShowAsync();
+        }
+
+        // this is where I put the commented code from their demo:
```
Can you clean this up? Either delete or implement if needed

UjfalusiAbel commented on Duo/ViewModels/ExerciseCreationViewModel.cs:
```diff
@@ -63,37 +63,38 @@ public ExerciseCreationViewModel()
+
+                SaveButtonCommand = new RelayCommand((_) => _ = CreateExercise());
+
+                /*ExerciseTypes = new ObservableCollection<string>
```
Can you clean up the unused lines? Either delete or implement

UjfalusiAbel commented on Duo/ViewModels/ExerciseCreationViewModel.cs:
```diff
@@ -63,37 +63,38 @@ public ExerciseCreationViewModel()
+                };*/
+                ExerciseTypes = new ObservableCollection<string>(Models.Exercises.ExerciseTypes.EXERCISE_TYPES);
+
+                /*Difficulties = new ObservableCollection<string>
```
Can you clean up the unused lines? Either delete or implement

UjfalusiAbel reviewed (changes requested):
small changes asked please, but it looks very good in general

ArturZsigmond committed:
fix: removed one line comment

ArturZsigmond committed:
fix: removed one line comment

ArturZsigmond committed:
fix: deleted old demo code

ArturZsigmond committed:
fix: added a method implementation

CristiSlv1 committed:
Add error handling in RoadmapSectionViewModel and View

Enhanced error handling in `RoadmapSectionViewModel.cs` by adding `try-catch` blocks during initialization and setup processes. Removed temporary data and improved the logic for populating quiz button templates with error handling.

Updated `RoadmapSectionView.xaml.cs` to attach an event handler for error messages from the ViewModel and added a `ShowErrorMessage` method for displaying error dialogs. Improved the `Quiz_Click` method to handle navigation errors gracefully.

CristiSlv1 committed:
Add error handling in quiz preview view models

Enhanced `RoadmapQuizPreviewViewModel.cs` with try-catch blocks for error handling during quiz order retrieval, UI updates, and initialization. Implemented a `RaiseErrorMessage` method for consistent error feedback.

Updated `RoadmapQuizPreview.xaml.cs` constructor to validate `DataContext` and handle errors during initialization. Improved `OpenQuizButton_Click` and `Load` methods to catch exceptions and provide user-friendly error messages.

CristiSlv1 committed:
Enhance error handling in RoadmapMainPageViewModel

Updated the RoadmapMainPageViewModel to include robust error handling during initialization and setup, utilizing try-catch blocks to manage exceptions and provide user feedback. Enhanced the OpenQuizPreview and StartQuiz methods with user-friendly error messages. Refactored the RoadmapMainPage.xaml.cs constructor and OnNavigatedTo method to ensure proper error handling and display of error messages. Overall, these changes improve application stability and user experience.

CristiSlv1 committed:
Improve error handling in MultipleChoiceExercise

Enhanced error handling in MultipleChoiceExerciseViewModel and
MultipleChoiceExercise.xaml.cs. Updated properties and methods to
use expression-bodied members and correct types. Added try-catch
blocks to manage exceptions during initialization, exercise loading,
and user interactions, ensuring better feedback for users.

CristiSlv1 committed:
Enhance error handling and refactor view model

Refactored `AssociationExerciseViewModel.cs` to use expression-bodied members and added try-catch blocks for improved error handling during initialization and method execution. The `GetExercise` method now validates exercise types and initializes `userAnswers` more safely.

Cleaned up using directives in `AssociationExercise.xaml.cs` and added error handling to various methods, ensuring graceful handling of exceptions during UI interactions. Introduced a centralized `ShowErrorMessage` method for consistent user feedback.

These changes improve the application's robustness and user experience by providing clearer error messages and preventing crashes.

CristiSlv1 committed:
Refactor error handling in exercise view models

Improved error handling and simplified property accessors in `FillInTheBlankExerciseViewModel.cs`. Enhanced constructor and methods to manage exceptions effectively.

In `FillInTheBlankExercise.xaml.cs`, added error handling for `DataContext`, introduced a dialog for error messages, and refactored the `ParseQuestion` method for better error management. Updated the `Send_Click` method to ensure user feedback on errors.

CristiSlv1 committed:
Add error handling to CreateAssociationExercise components

Enhanced robustness in CreateAssociationExerciseViewModel.cs by adding try-catch blocks for error handling in key methods, including the constructor, AddNewAnswer, CreateExercise, and GenerateAnswerList. Updated the Answer class to include error handling in its property setter and constructor.

In CreateAssociationExercise.xaml.cs, improved initialization error handling in the constructor and added a ShowErrorMessage method to display error dialogs, enhancing user feedback during control initialization.

CristiSlv1 committed:
Add error handling to exercise view models and UI

Enhanced error handling in CreateFillInTheBlankExerciseViewModel.cs by adding try-catch blocks to various methods, ensuring exceptions are caught and appropriate error messages are raised. Updated the constructor to throw an ArgumentNullException for null parentViewModel and added checks for maximum answers in AddNewAnswer.

In CreateFillInTheBlankExercise.xaml.cs, improved initialization error handling by validating the DataContext and added a method to display error messages in a dialog, enhancing UI robustness.

CristiSlv1 committed:
Add error handling to flashcard exercise components

Enhance `CreateFlashcardExerciseViewModel` with try-catch blocks in the constructor, `Answer` property setter, and `CreateExercise` method to manage exceptions and raise error messages.

Update `CreateFlashcardExercise` to include error handling during initialization, with new methods for displaying error messages in a dialog asynchronously, improving user feedback on errors.

CristiSlv1 committed:
Add error handling to exercise view models and UI

Enhanced error handling in `CreateMultipleChoiceExerciseViewModel.cs` by adding try-catch blocks to methods and properties, improving robustness. Marked `parentViewModel` as `readonly` and updated methods to check for maximum answers and handle exceptions.

Updated `CreateMultipleChoiceExercise.xaml.cs` constructor to include error handling for DataContext initialization and added methods for displaying error messages in a dialog, improving user feedback during initialization errors.

CristiSlv1 committed:
Refactor error handling in view model and pages

Improved error handling and encapsulation in `QuizExamViewModel.cs` by modifying properties to have private setters and adding exception handling in various methods. Updated constructors and navigation methods in `ExamPage.xaml.cs`, `QuizPage.xaml.cs`, and `QuizPreviewPage.xaml.cs` to include error handling, enhancing user feedback and robustness across the application.

CristiSlv1 committed:
Add error handling to AnswerFeedbackPopup

Enhanced error handling in AnswerFeedbackPopup.xaml.cs by wrapping key methods in try-catch blocks. Introduced a new asynchronous method, ShowErrorMessage, to display error dialogs for initialization, feedback display, and dialog closure errors, improving user experience and robustness.
